test/browser/spec/ol/style/icon.test.js

```diff
@@ -435,24 +470,60 @@ describe('ol.style.Icon', function () {
       });
       expect(iconStyle.getHeight()).to.eql(20);
     });
-    it('setHeight updates the height', function () {
+    it('setHeight updates the height, and scale when image is loaded', function (done) {
       const iconStyle = new Icon({
-        src,
+        src: img,
         height: 20,
       });
       expect(iconStyle.getHeight()).to.eql(20);
+      expect(iconStyle.getScale()).to.eql(1);
       iconStyle.setHeight(200);
       expect(iconStyle.getHeight()).to.eql(200);
+      expect(iconStyle.getScale()).to.eql(1);
+
+      const iconImage = iconStyle.iconImage_;
+      iconImage.addEventListener('change', function () {
+        expect(iconStyle.getHeight()).to.eql(200);
+        expect(iconStyle.getScale()).to.eql([50, 50]);
+        iconStyle.setHeight(20);
+        expect(iconStyle.getHeight()).to.eql(20);
+        expect(iconStyle.getScale()).to.eql([5, 5]);
+        done();
+      });
+      iconStyle.load();
     });
-    it('setScale updates the width and height', function (done) {
+    it('setWidth and setHeight update scale independently', function (done) {
+      const iconStyle = new Icon({
+        src: img,
+        width: 9,
+        height: 20,
+      });
+      const iconImage = iconStyle.iconImage_;
+      iconImage.addEventListener('change', function () {
+        expect(iconStyle.getWidth()).to.eql(9);
+        expect(iconStyle.getHeight()).to.eql(20);
+        expect(iconStyle.getScale()).to.eql([3, 5]);
+        iconStyle.setWidth(30);
+        expect(iconStyle.getWidth()).to.eql(30);
+        expect(iconStyle.getHeight()).to.eql(20);
+        expect(iconStyle.getScale()).to.eql([10, 5]);
+        iconStyle.setHeight(200);
+        expect(iconStyle.getWidth()).to.eql(30);
+        expect(iconStyle.getHeight()).to.eql(200);
+        expect(iconStyle.getScale()).to.eql([10, 50]);
+        done();
+      });
+      iconStyle.load();
+    });
+    it('setScale clears the width and height', function (done) {
       const iconStyle = new Icon({
         src: img,
       });
       const iconImage = iconStyle.iconImage_;
       iconImage.addEventListener('change', function () {
         iconStyle.setScale(2);
-        expect(iconStyle.getWidth()).to.eql(6);
-        expect(iconStyle.getHeight()).to.eql(8);
+        expect(iconStyle.getWidth()).to.eql(undefined);
+        expect(iconStyle.getHeight()).to.eql(undefined);
```

Comment:
This is a change in behaviour which should be mentioned in the upgrade notes.

Reply:
Done, also fixed some timing issues and now using imageState to avoid initializeImage being called sooner than necessary.  The extra listener was detected in one renderer test.

Reply:
LIke I said above, the latter looks suspicious to me.

Reply:
> LIke I said above, the latter looks suspicious to me.

The former was wrong.  `setScale` should not have been setting `width` or `height` (especially as they were not set before that).